Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjection.php

```diff
@@ -156,10 +155,14 @@ public function markStale(): void
 
     private function truncateDatabaseTables(): void
     {
-        $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->node());
-        $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->hierarchyRelation());
-        $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->referenceRelation());
-        $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->dimensionSpacePoints());
+        try {
+            $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->node());
+            $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->hierarchyRelation());
+            $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->referenceRelation());
+            $this->dbal->executeQuery('TRUNCATE table ' . $this->tableNames->dimensionSpacePoints());
+        } catch (DbalException $e) {
+            throw new \RuntimeException(sprintf('Failed to truncate database tables for projection %s: %s', self::class, $e->getMessage()), 1716478318, $e);
+        }
```

Comment:
jip this case actually happened to me https://github.com/neos/neos-development-collection/issues/5008 id say we can ignore the table does not exists exceptions and move to the next table?

`flow cr:setup --reset-projections` was intended to be the most bruteforce way to get the sql shema fixed. It should not die because of this ;)

Reply:
it would still die, but with a different exception.. I'll have to think about whether we really would want to silently ignore missing tables here (my gut feeling is: no). But in any case: I tried not to change any behavior (except for mini bugfixes) with this PR

Reply:
I think `./flow cr:setup --reset-projections` just invoke the setup _first_ – that should solve things, no?

Reply:
=> #5089

Reply:
thanks and sorry for exploding the discussion here :O